sklearn/preprocessing/data.py

```diff
@@ -1179,9 +1180,39 @@ def powers_(self):
         combinations = self._combinations(self.n_input_features_, self.degree,
                                           self.interaction_only,
                                           self.include_bias)
-        return np.vstack(np.bincount(c, minlength=self.n_input_features_)
+        return np.vstack(bincount(c, minlength=self.n_input_features_)
                          for c in combinations)
 
+    def get_feature_names(self, input_features=None):
+        """
+        Return feature names for output features
+
+        Parameters
+        ----------
+        input_features : list of string, length n_features, optional
+            String names for input features if available. By default,
+            "x0", "x1", ... "xn_features" is used.
+
+        Returns
+        -------
+        output_feature_names : list of string, length n_output_features
+
+        """
+        powers = self.powers_
+        if input_features is None:
+            input_features = ['x%d' % i for i in range(len(powers))]
```

Comment:
I concede you are right, and that we're already using the fact that this is the incorrect shape in the loop below. Of course, this estimator is only useful when there are at least as many output as input features, so the code's going to work here.

But good catch. Should be fixed here to use `powers.shape[1]`; documentation of the attribute should be fixed; and the attribute should be tested directly in a unit test.